ui_test/src/comments.rs

```diff
@@ -140,36 +208,141 @@ impl Comments {
                 );
                 this.error_pattern = Some((s.trim().to_string(), l));
             }
-            if let Some(captures) = error_pattern_regex.captures(line) {
-                // FIXME: check that the error happens on the marked line
-                let matched = captures["text"].trim().to_string();
+        }
+        this
+    }
 
-                let revision = captures.name("revision").map(|rev| rev.as_str().to_string());
+    fn parse_command_with_args(&mut self, command: &str, args: &str, path: &Path, l: usize) {
+        checked!(path, l);
+        match command {
+            "revisions" => {
+                check!(self.revisions.is_none(), "cannot specifiy revisions twice");
+                self.revisions = Some(args.split_whitespace().map(|s| s.to_string()).collect());
+            }
+            "compile-flags" => {
+                self.compile_flags.extend(args.split_whitespace().map(|s| s.to_string()));
+            }
+            "rustc-env" =>
+                for env in args.split_whitespace() {
+                    let (k, v) = unwrap!(
+                        env.split_once('='),
+                        "environment variables must be key/value pairs separated by a `=`"
+                    );
+                    self.env_vars.push((k.to_string(), v.to_string()));
+                },
+            "normalize-stderr-test" => {
+                let (from, to) =
+                    unwrap!(args.split_once("->"), "normalize-stderr-test needs a `->`");
+                let from = from.trim().trim_matches('"');
+                let to = to.trim().trim_matches('"');
+                let from = unwrap!(Regex::new(from).ok(), "invalid regex");
+                self.normalize_stderr.push((from, to.to_string()));
+            }
+            "error-pattern" => {
+                check!(
+                    self.error_pattern.is_none(),
+                    "cannot specifiy error_pattern twice, previous: {:?}",
+                    self.error_pattern
+                );
+                self.error_pattern = Some((args.trim().to_string(), l));
+            }
+            // Maybe the user just left a comment explaining a command without arguments
+            _ => self.parse_command(command, path, l),
```

Comment:
...? This is odd.

I think what I would have expected is a parser that takes things from the line. Eagerly finding the `:` and this entire with/without args split doesn't look like normal parsing.^^

Reply:
yea... it felt convenient to do it this way ^^ but I can do a proper token munching parser instead.